test/parallel_api/algorithm/alg.sorting/alg.binary.search/binary.search/binary_search.pass.cpp

```diff
@@ -86,10 +87,33 @@ int main()
     if((res_2[0] != true) || (res_2[1]!= true) || (res_2[2] != false) || (res_2[3] != false) || (res_2[4] != false))
         correctness_flag = false;
     
-    if(correctness_flag == true)
-        std::cout << "done" << std::endl;
-    else
-        std::cout << "Values do not match." << std::endl;
-    
+    if(correctness_flag != true)
+        std::cout << "binary_search on device FAIL." << std::endl;
+}
+#endif
+
+bool test_on_host()
+{
+    int key[10] = {0, 2, 2, 2, 3, 3, 3, 3, 6, 6};
+    int val[5] = {0, 2, 4, 7, 6};
+    int res[5];
+  
+     // call algorithm
+     oneapi::dpl::binary_search(oneapi::dpl::execution::par, std::begin(key), std::end(key), std::begin(val), std::end(val), std::begin(res), std::less<int>());
```

Comment:
To check algorithm with different policies we often use `invoke_on_all_policies` fucntion from `support/utils.h` (example: https://github.com/oneapi-src/oneDPL/blob/main/test/parallel_api/algorithm/alg.modifying.operations/fill.pass.cpp#L93).

You can try this function too. It will save you from using different functions for host and device and the code will be reduced. I can share all capabilities of `support/utils.h` if you want. Then you can choose something from the file to make tests more consistent and to cover more cases.

Reply:
Updates of the tests to use `invoke_on_all_policies` will be done in a separate PR.